lib/manageiq/providers/nuage/engine.rb

```diff
@@ -11,6 +11,12 @@ class Engine < ::Rails::Engine
           app.config.paths["config/secrets"] << root.join("config", "secrets.yml").to_s
         end
 
+        initializer 'manageiq-providers-nuage.configure_autoloader' do
+          # The following file should not be eager loaded as it depends on being installed, which isn't possible on macs and isn't done by default
+          # in ci for other plugins, core, etc.
+          Rails.autoloaders.main.ignore(root.join("app/models/manageiq/providers/nuage/network_manager/event_catcher/messaging_handler.rb"))
```

Comment:
I'm wondering if I should do a `require 'qpid_proton' rescue... ^^^` that line here so we don't need to require the handler in the spec.  Autoload from within the engine should work if qpid_proton is installed.

Reply:
The problem with this is that all production appliances/pods where qpid_proton is installed will eager require qpid_proton when this engine is initialized.

Reply:
In openstack we exclude it from the rspec configuration if qpid_proton isn't loaded could we do that here as well?

https://github.com/ManageIQ/manageiq-providers-openstack/blob/master/spec/spec_helper.rb#L11-L13

Reply:
yes, that works, thank you